docs/design.md

```diff
@@ -1,87 +1,133 @@
+    os-->>n: 
+    n-->>k: Kubelet update Node.volumesInUse
+    deactivate n
+    
+    note over k: Pod 'Running'
+    k-->>o: Stateful workload running
+    deactivate k
+```
 
-## Requirements
+## CSI Driver Design Requirements
 
 ### Idempotency
 
-All CSI driver calls should be idempotent. A CSI method call with the same parameters must always return the same result. It&#39;s task of CSI driver to ensure that. Examples:
+All CSI calls should be idempotent. The CSI plugin must ensure that a CSI call with the same parameters will always return the same result. Examples:
 
 - CreateVolume call must first check that the requested EBS volume has been already provisioned and return it if so. It should create a new volume only when such volume does not exist.
 - ControllerPublish (=i.e. attach) does not do anything and returns &quot;success&quot; when given volume is already attached to requested node.
 - DeleteVolume does not do anything and returns success when given volume is already deleted (i.e. it does not exist, we don&#39;t need to check that it had existed and someone really deleted it)
 
-Note that it&#39;s task of the CSI driver to make these calls idempotent if related AWS API call is not.
+Note that it&#39;s task of the ebs-plugin to make these calls idempotent even if the related AWS API call is not.
 
 ### Timeouts
 
-gRPC always passes a timeout together with a request. After this timeout, the gRPC client call actually returns. The server (=CSI driver) can continue processing the call and finish the operation, however it has no means how to inform the client about the result.
+gRPC always passes a timeout together with a request. After this timeout, the gRPC client call actually returns. The server (i.e. ebs-plugin) can continue processing the call and finish the operation, however it has no means how to inform the client about the result.
 
-Kubernetes will retry failed calls, usually after some exponential backoff. Kubernetes heavily relies on idempotency here - i.e. when the driver finished an operation after the client timed out, the driver will get the same call again and it should return success/error based on success/failure of the previous operation.
+Kubernetes sidecars will retry failed calls after exponential backoff. These sidecars rely on idempotency here - i.e. when ebs-plugin finished an operation after the client timed out, the ebs-plugin will get the same call again, and it should return success/error based on success/failure of the previous operation.
 
 Example:
 
-1. Kubernetes calls ControllerPublishVolume(vol1, nodeA) ), i.e. &quot;attach vol1 to nodeA&quot;.
-2. The CSI driver checks vol1 and sees it&#39;s not attached to nodeA yet. It calls AttachVolume(vol1, nodeA).
-3. The attachment takes a long time, Kubernetes times out.
-4. Kubernetes sleeps for some time.
+1. csi-attacher calls ControllerPublishVolume(vol1, nodeA) ), i.e. &quot;attach vol1 to nodeA&quot;.
```

Comment:
Keep the wording Kubernetes, if you want to call out which sidecar makes the request you could do `Kubernetes (via the csi-attacher sidecar) calls ...`